compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -682,16 +682,304 @@ abstract class GenJSCode extends plugins.PluginComponent
             "repeated parameters.")
       }
```

Comment:
The above implementation restriction should also be lifted (for free) by this implementation. So this should disappear too, and there should be tests instead.